src/Aspire.Hosting.Azure/AzureConstructResource.cs

```diff
@@ -119,78 +120,129 @@ public static IResourceBuilder<T> ConfigureConstruct<T>(this IResourceBuilder<T>
+            parameter = new BicepParameter(parameterName, typeof(string))
+            {
+                IsSecure = parameterResourceBuilder.Resource.Secret
+            };
+            construct.Add(parameter);
         }
-        else
+
+        return parameter;
+    }
+
+    /// <summary>
+    /// TODO
+    /// </summary>
+    /// <param name="outputReference"></param>
+    /// <param name="construct"></param>
+    /// <param name="parameterName"></param>
+    /// <returns></returns>
+    [SuppressMessage("ApiDesign", "RS0026:Do not add multiple public overloads with optional parameters",
+        Justification = "The 'this' arguments are mutually exclusive")]
+    public static BicepParameter AsBicepParameter(this BicepOutputReference outputReference, ResourceModuleConstruct construct, string? parameterName = null)
+    {
+        ArgumentNullException.ThrowIfNull(outputReference);
+        ArgumentNullException.ThrowIfNull(construct);
+
+        parameterName ??= outputReference.Name;
+
+        construct.Resource.Parameters[parameterName] = outputReference;
+
+        var parameter = construct.GetParameters().FirstOrDefault(p => p.ResourceName == parameterName);
+        if (parameter is null)
         {
-            var parameter = new Parameter(parameterName, isSecure: parameterResourceBuilder.Resource.Secret);
-            resource.AssignProperty(propertySelector, parameter);
+            parameter = new BicepParameter(parameterName, typeof(string));
+            construct.Add(parameter);
         }
+
+        return parameter;
     }
 
     /// <summary>
-    /// Assigns an Aspire Bicep output reference to an Azure construct resource.
+    /// Assigns an Aspire parameter resource to an Azure construct resource.
     /// </summary>
     /// <typeparam name="T">Type of the CDK resource.</typeparam>
     /// <param name="resource">The CDK resource.</param>
     /// <param name="propertySelector">Property selection expression.</param>
+    /// <param name="parameterResourceBuilder">Aspire parameter resource builder.</param>
     /// <param name="parameterName">The name of the parameter to be assigned.</param>
-    /// <param name="outputReference">Aspire parameter resource builder.</param>
     [System.Diagnostics.CodeAnalysis.SuppressMessage("ApiDesign", "RS0026:Do not add multiple public overloads with optional parameters", Justification = "<Pending>")]
-    public static void AssignProperty<T>(this Resource<T> resource, Expression<Func<T, object?>> propertySelector, BicepOutputReference outputReference, string? parameterName = null) where T : notnull
+    [Obsolete("This API is no longer supported and always throws.")]
+    public static void AssignProperty<T>(this global::Azure.Provisioning.Primitives.Resource resource, System.Linq.Expressions.Expression<Func<T, object?>> propertySelector, IResourceBuilder<ParameterResource> parameterResourceBuilder, string? parameterName = null) where T : notnull
```

Comment:
@mitchdenny @davidfowl - what do you think about this? It is trying to keep the `AssignProperty` method with an obsoletion so any callers of it get a warning that it is no longer supported.

It may not be worth it, and maybe we should just delete it?

Reply:
I'm gong to remove this because it isn't the exact same API. The exisitng API can't be built because it is based on a Type that no longer exists in the new CDK. So I think we should remove these APIs altogether.